Add EasyBind.concat(List<ObservableList<T>>)

maul-esel opened the pull request:
Creates a new list that combines the values of the given lists. Unlike `FXCollections.concat()`, updates to the source lists propagate to the combined list.

I needed this for a project of mine, so I hacked it together. Since it might be useful to others as well, I thought I'd submit it here. If there are any changes you'd like me to make, let me know.

Commit:
Add EasyBind.concat(List<ObservableList<T>>)

Creates a new list that combines the values of the given
lists. Unlike FXCollections.concat(), updates to the source
lists propagate to the combined list.

TomasMikula commented:
Thanks for the PR!

OMG, I didn't expect `FXCollections.concat()` to be so silly!

I can see how your implementation would fail on

```java
concat(list, list);
```

Some tests would be nice.

Commit:
Fix compiler errors

Commit:
add some tests for concatenating lists

maul-esel commented:
I added some tests, hope they're ok. Otherwise let me know.

I also added a short test for `concat(list, list);` and so far it seems to work. What part exactly did you expect to fail? (maybe I didn't cover that)

TomasMikula commented:
You did not test that the published change events are correct.

N247S commented:
Just as an idea, wouldn't it be better (for boilersplate's sake) to switch out the `sourceLists` field in `ConcatList` with a plain array? As it's immutable, and operations performed on the List are easily done as array operations (This would also means the `concat` method in `EasyBind` can be merged into a single varargs method).

Second (for performance sake), would it be an idea to cage the overal size? As it can be detected when the underlaying lists are changing, it would easy to refresh/recalculate the overal size (or at least set a flag that the current size is invalid), whichever way is prefered.

tobiasdiez commented:
Since this project sadly doesn't seem to be active any more #16, I've created a fork and continued development there:  https://github.com/tobiasdiez/EasyBind.

I liked your changes in this PR, and thus included them in the forked repo. Hope this is OK for you.
